geospaas/utils/processing_base_command.py

```diff
@@ -0,0 +1,153 @@
+import os
+import glob
+from datetime import datetime as dt
+
+from django.db.models import Q
+from django.core.management.base import BaseCommand
+from django.contrib.gis.geos import GEOSGeometry, Polygon, GEOSException
+from django.contrib.gis.gdal import GDALException
+
+from geospaas.catalog.models import Dataset
+
+def valid_date(s):
+    ''' Test input datestring '''
+    try:
+        return dt.strptime(s, "%Y-%m-%d")
+    except ValueError:
+        msg = "Not a valid date: '{0}'.".format(s)
+        raise TypeError(msg)
```

Comment:
It maybe would be better to use `CommandError` here: https://docs.djangoproject.com/en/2.0/howto/custom-management-commands/ (very bottom of the page)